logdevice/common/test/NodesConfigurationPollerTest.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#include "logdevice/common/NodesConfigurationPoller.h"
+
+#include <algorithm>
+
+#include <folly/Random.h>
+#include <gtest/gtest.h>
+
+#include "logdevice/common/ObjectPoller.h"
+#include "logdevice/common/RandomNodeSelector.h"
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/nodes/NodesConfigurationCodec.h"
+#include "logdevice/common/test/TestUtil.h"
+#include "logdevice/common/toString.h"
+
+using namespace facebook::logdevice;
+
+namespace {
+
+class NodesConfigurationPollerExtended : public NodesConfigurationPoller {
```

Comment:
Typically we call them `MockNodesConfigurationPoller` instead.

Reply:
done